obs-to-maven

```diff
@@ -33,39 +33,50 @@ import zlib
 
 
 class Rpm:
-    def __init__(self, location, mtime):
+    def __init__(self, location, mtime, name, version):
         self.path = location
         self.mtime = mtime
         self.name = location[location.find('/') + 1:]
+        self.pkgname = name
+        self.version = version
 
 class Repo:
     def __init__(self, uri, project, repository):
         self.uri = uri
-        self.project = project
+        self.project = project.replace(':', ':/')
         self.repository = repository
         self._rpms = None
 
     def find_primary(self):
         ns = {'repo': 'http://linux.duke.edu/metadata/repo', 'rpm': 'http://linux.duke.edu/metadata/rpm'}
         f = urllib.request.urlopen(
-            "{}/repositories/{}/{}/repodata/repomd.xml".format(self.uri, self.project, self.repository)
+            "{}/{}/{}/repodata/repomd.xml".format(self.uri, self.project, self.repository)
         )
         doc = ET.fromstring(f.read())
         primary_href = doc.find("./repo:data[@type='primary']/repo:location", ns).get("href")
-        return "{}/repositories/{}/{}/{}".format(self.uri, self.project, self.repository, primary_href)
+        return "{}/{}/{}/{}".format(self.uri, self.project, self.repository, primary_href)
 
     def parse_primary(self):
         ns = {"c": "http://linux.duke.edu/metadata/common", "rpm":"http://linux.duke.edu/metadata/rpm"}
         f = urllib.request.urlopen(self.find_primary())
         primary_xml = zlib.decompress(f.read(), 16 + zlib.MAX_WBITS)
         doc = ET.fromstring(primary_xml)
 
-        self._rpms = [
-            Rpm(n.find("c:location", ns).get("href"), int(n.find("c:time", ns).get("file")))
+        all_rpms = [
+            Rpm(n.find("c:location", ns).get("href"),
+                int(n.find("c:time", ns).get("file")),
+
+                n.find("c:name", ns).text,
+                "{}-{}".format(n.find("c:version", ns).get("ver"), n.find("c:version", ns).get("rel")))
             for n
             in doc.findall(".//c:package", ns)
             if n.find("c:arch", ns).text in ["x86_64", "noarch"]
         ]
+        latest_rpms = {}
+        for rpm in all_rpms:
+            if not latest_rpms.get(rpm.pkgname) or latest_rpms[rpm.pkgname].version < rpm.version:
```

Comment:
A normal compare is not working for PRM versions.
Or does here happen something special I am not aware of?

Reply:
why wouldn't it? it's just comparing versions+build numbers and should follow lexicographical order, unless there is some case I overlooked

Reply:
But this is not how RPM versions are compared. The good thing is, that RPM provide a python module which can be used.
This seems to be a good page with some examples and hints https://stackoverflow.com/questions/3206319/how-do-i-compare-rpm-versions-in-python

Reply:
How about using the rpm function and something like the regexp based fallback described in https://stackoverflow.com/a/42967591/4140893 ? I can't count on the RPM python binding since those are not available on macos.

Reply:
Seems it must be good enough when rpm bindings are not available